datastream-server-api/src/main/java/com/linkedin/datastream/server/callbackstatus/CallbackStatusWithComparableOffsets.java

```diff
@@ -0,0 +1,93 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server.callbackstatus;
+
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.PriorityQueue;
+import java.util.Queue;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Helper class to store the callback status of the inflight events with comparable offsets.
+ *
+ * @param <T> Type of the comparable checkpoint object internally used by the connector.
+ */
+public class CallbackStatusWithComparableOffsets<T extends Comparable<T>> extends CallbackStatus<T> {
+
+  private static final Logger LOG = LoggerFactory.getLogger(CallbackStatusWithComparableOffsets.class);
+
+  private T _highWaterMark = null;
```

Comment:
since you are moving the code, plz take care of this as well.  Organize the code in the following order.

static final 
final
regular objects.
method definition.

We have tried to do this pattern whereever possible in this github repo.